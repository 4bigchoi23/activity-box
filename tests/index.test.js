const nock = require('nock')
const { createBody, updateGist } = require('../')

const events = [{
  type: 'IssuesEvent',
  repo: { name: 'clippy/take-over-github' },
  payload: { action: 'opened', issue: { number: 1 } }
}, {
  type: 'IssueCommentEvent',
  repo: { name: 'clippy/take-over-github' },
  payload: { action: 'closed', issue: { number: 1 } }
}, {
  type: 'PullRequestEvent',
  repo: { name: 'clippy/take-over-github' },
  payload: { action: 'closed', pull_request: { number: 2, merged: true } }
}, {
  type: 'PullRequestEvent',
  repo: { name: 'clippy/take-over-github' },
  payload: { action: 'closed', pull_request: { number: 3, merged: false } }
}]

describe('activity-box', () => {
  beforeEach(() => {
    nock('https://api.github.com')
      .get('/users/clippy/events/public?per_page=100')
      .reply(200, events)
  })

  describe('createBody', () => {
    it('returns the expected string', async () => {
      const actual = await createBody()
      expect(actual).toMatchSnapshot()
    })
  })

  afterEach(() => {
    delete process.env.GITHUB_TOKEN
    delete process.env.GITHUB_USERNAME
    delete process.env.GIST_ID
  })
})