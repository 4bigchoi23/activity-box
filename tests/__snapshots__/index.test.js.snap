// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity-box createBody returns the expected string 1`] = `
"❗️ Opened issue #1 in clippy/take-over-github
🗣 Commented on #1 in clippy/take-over-github
🎉 Merged PR #2 in clippy/take-over-github
❌ Closed PR #3 in clippy/take-over-github"
`;
